pkgs/top-level/darwin-aliases.nix

```diff
@@ -163,7 +154,8 @@ stubs
   ### S ###
 
   stdenvNoCF =
-    lib.warn "darwin.stdenvNoCF: use `stdenv` or `stdenvNoCC`; this will be removed in 25.11"
+    lib.warnOnInstantiate
+      "darwin.stdenvNoCF: use `stdenv` or `stdenvNoCC`; this will be removed in 25.11"
```

Comment:
Correctly warns the first time `darwin.stdenvNoCF.mkDerivation` is used:

```
nix-repl> :b darwin.stdenvNoCF.mkDerivation { name = "test"; }
evaluation warning: darwin.stdenvNoCF: use `stdenv` or `stdenvNoCC`; this will be removed in 25.11
error: builder for '/nix/store/wn6zg4zh9as201sryrf4v15v3hra8j1v-test.drv' failed with exit code 1;
       last 2 log lines:
       > Running phase: unpackPhase
       > variable $src or $srcs should point to the source
       For full logs, run:
         nix log /nix/store/wn6zg4zh9as201sryrf4v15v3hra8j1v-test.drv
[0 built (1 failed)]
```